nix/modules/scripts.nix

```
@@ -57,7 +49,8 @@
           git add "$VERSIONS_DIR"/flake.lock
         fi
 
-        if [[ "$VERSIONS_DIR" == "$DEFAULT_VERSIONS_DIR" ]]; then
+        # Want to update the root flake.lock if we're updating the version that is currently the default.
+        if grep -q "dir=$VERSIONS_DIR" flake.nix; then
```

i'd prefer matching the line more specifically:

```suggestion
        if grep -qE "versions.url = \".+\?dir=''${VERSIONS_DIR}\"" flake.nix; then
```

---

This regex doesn't work, I think because of the `/` in the substituted VERSIONS_DIR needing escaping

---

Oh never mind, this is my grep being different - this is good

---

With some minor adjustments, to escape the `.` in `versions.url` and to remove the `''` before the `${`

---

the `''` escape the `${` in nix strings and they need to be removed for trying it out in your shell